src/main/java/com/uber/jenkins/phabricator/BuildResultProcessor.java

```diff
@@ -0,0 +1,205 @@
+            } else {
+                logger.info(LOGGING_TAG, "No Uberalls backend configured, skipping...");
+            }
+        } else {
+            logger.info(LOGGING_TAG, "No line coverage found, skipping...");
+        }
+
+    }
+
+    public void processBuildResult(boolean commentOnSuccess, boolean commentWithConsoleLinkOnFailure) {
+        commenter.processBuildResult(commentOnSuccess, commentWithConsoleLinkOnFailure, runHarbormaster);
+    }
+
+    public void processRemoteComment(String commentFile, String commentSize) {
+        RemoteCommentFetcher commentFetcher = new RemoteCommentFetcher(workspace, logger, commentFile, commentSize);
+        try {
+            String customComment = commentFetcher.getRemoteComment();
+            commenter.addUserComment(customComment);
+        } catch (InterruptedException e) {
+            e.printStackTrace(logger.getStream());
+        } catch (IOException e) {
+            e.printStackTrace(logger.getStream());
+        }
+    }
+
+    public void sendComment(boolean commentWithConsoleLinkOnFailure) {
+        if (!commenter.hasComment()) {
+            return;
+        }
+
+        if (commentWithConsoleLinkOnFailure && buildResult.isWorseOrEqualTo(hudson.model.Result.UNSTABLE)) {
+            commenter.addBuildFailureMessage();
+        } else {
+            commenter.addBuildLink();
+        }
+
+        new PostCommentTask(logger, diffClient, diff.getRevisionID(false), commenter.getComment(), commentAction).run();
+    }
+
+    public boolean processHarbormaster() {
+        final boolean harbormasterSuccess = buildResult.isBetterOrEqualTo(Result.SUCCESS);
+
+        if (runHarbormaster) {
+            logger.info("harbormaster", "Sending Harbormaster BUILD_URL via PHID: " + phid);
+            Task.Result sendUriResult = new SendHarbormasterUriTask(logger, diffClient, phid, buildUrl).run();
+
+            if (sendUriResult != Task.Result.SUCCESS) {
+                logger.info(LOGGING_TAG, "Unable to send BUILD_URL to Harbormaster");
+            }
+
+            if (getCoverage() != null) {
+                logger.info(
+                        LOGGING_TAG,
+                        String.format("Publishing coverage data to Harbormaster for %d files.", getCoverage().size())
+                );
+            }
+
+            logger.info(
+                    LOGGING_TAG,
+                    String.format("Sending build result to Harbormaster with PHID %s, success: %s",
+                            phid,
+                            harbormasterSuccess
+                    )
+            );
+
+            Task.Result result = new SendHarbormasterResultTask(logger, diffClient, phid, harbormasterSuccess, getCoverage()).run();
+            if (result != Task.Result.SUCCESS) {
+                logger.info(LOGGING_TAG, "Unable to post to harbormaster");
```

Comment:
Hmm.. doesn't it make more sense for this log statement to live inside the task itself?